InnoGotchiBackend/InnnoGotchi.DAL/Respositories/PetRepository.cs

```diff
@@ -4,42 +4,65 @@
 
 namespace InnnoGotchi.DAL.Respositories
 {
-    public class PetRepository : InnoGotchiContextHandler, IRepository<Pet>
+    /// <summary>
+    /// Represents repository that has full access to pet entities in database
+    /// </summary>
+    public class PetRepository : IRepository<Pet>
     {
-        public PetRepository(InnoGotchiContext context) : base(context) { }
+        private InnoGotchiContext _context;
+        public PetRepository(InnoGotchiContext context)
+        {
+            _context = context;
+        }
+
+        public bool Contains(Func<Pet, bool> predicate)
+        {
+            Pet? pet = FirstOrDefault(predicate);
+            if (pet == null)
+                return false;
+            else
+                return true;
+        }
 
         public void Create(Pet item)
         {
             _context.Pets.Add(item);
         }
 
-        public void Delete(int id)
+        public bool Delete(int id)
         {
-            Pet? pet = _context.Pets.FirstOrDefault(p => p.Id == id);
+            Pet? pet = Get(id);
             if (pet != null)
             {
                 _context.Pets.Remove(pet);
+                return true;
             }
+            return false;
+        }
+
+        public IQueryable<Pet> FindAll(Func<Pet, bool> predicate)
+        {
+            return GetAll().Where(predicate).AsQueryable();
         }
 
-        public IEnumerable<Pet> Find(Func<Pet, bool> predicate)
+        public Pet? FirstOrDefault(Func<Pet, bool> predicate)
         {
-            return _context.Pets.Where(predicate);
+            return GetAll().FirstOrDefault(predicate);
         }
 
         public Pet? Get(int id)
         {
-            return _context.Pets.FirstOrDefault(p => p.Id == id);
+            return FirstOrDefault(p => p.Id == id);
         }
 
-        public IEnumerable<Pet> GetAll()
+        public IQueryable<Pet> GetAll()
```

Comment:
GetAll - naming for real collections, not for IQueryable

Reply:
How to name this method?